core/primitives/src/challenge.rs

```diff
@@ -51,13 +51,14 @@ pub struct ChunkProofs {
     /// Merkle proof of inclusion of this chunk.
     pub merkle_proof: MerklePath,
     /// Invalid chunk in an encoded form or in a decoded form.
-    pub chunk: MaybeEncodedShardChunk,
+    pub chunk: Box<MaybeEncodedShardChunk>,
```

Comment:
Hmm... naive question but could you please explain why putting this in a Box helps?  Is it to move the allocation to the heap?

Also should we update the comment to add a brief explanation as to why a box is being used.

Reply:
Correct. `MaybeEncodedShardChunk` is a, if I remember correctly, 500 byte ADT. Which makes this containing structure even larger. Every time this structure is moved around (which in Rust can happen frequently) all of these bytes are memcpyd. With a `Box` this structure becomes a 56 byte one, resulting in much less memory traffic overall.